gcc/rust/util/rust-attributes.cc

```diff
@@ -223,27 +223,30 @@ check_doc_attribute (const AST::Attribute &attribute)
       }
     }
 }
-
-static bool
-is_proc_macro_type (const AST::Attribute &attribute)
+tl::optional<std::string>
+Attributes::is_proc_macro_type (const AST::Attribute &attribute)
```

Comment:
You're changing the semantic of the function, either you need a new one (two different functions) or you need to rename this one, we're not deciding if an attribute is or is not a proc macro anymore but returning the kind of proc macro with your new code.

I'd keep two different functions, even if one is calling the other and simply using `tl::optional::has_value()`